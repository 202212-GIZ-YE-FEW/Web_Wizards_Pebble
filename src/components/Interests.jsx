import SecondaryTitle from "./SecondaryTitle";
import SingleInterest from "./SingleInterest";

const Interests = ({ t }) => {
    const interests = [
        "noPoverty",
        "goodHealth",
        "qualityEducation",
        "genderEquality",
        "cleanWater",
        "affordableEnergy",
        "decentWork",
        "innovation",
        "reducedInequalities",
        "sustainableCities",
        "responsibleConsumption",
        "lifeBelowWater",
        "lifeOnLand",
        "peaceJustice",
    ];
    return (
        <div className='mt-10 '>
            <SecondaryTitle text={t("yourInterests")} />
            <div className='flex flex-wrap justify-evenly'>
                {interests.map((interest) => (
                    <SingleInterest key={interest} text={t(interest)} />
                ))}
            </div>
        </div>
    );
};

export default Interests;
