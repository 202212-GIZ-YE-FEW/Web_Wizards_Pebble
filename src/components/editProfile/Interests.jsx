import { useState } from "react";
import { useTranslation } from "react-i18next";

import SecondaryTitle from "./SecondaryTitle";
import SingleInterest from "./SingleInterest";
const Interests = () => {
    const { t } = useTranslation("interests");

    const interests = [
        "no_poverty",
        "zero_hunger",
        "good_health",
        "quality_education",
        "gender_equality",
        "clean_water",
        "affordable_energy",
        "decent_work",
        "innovation",
        "reduced_inequalities",
        "sustainable_cities",
        "responsible_consumption",
        "climate_action",
        "life_below_water",
        "life_on_land",
        "peace_justice",
    ];

    const [clickedInterests, setClickedInterests] = useState([]);

    const handleInterestClick = (interest) => {
        if (clickedInterests.includes(interest)) {
            setClickedInterests(clickedInterests.filter((i) => i !== interest));
        } else {
            setClickedInterests([...clickedInterests, interest]);
        }
    };
    // Here is the clolected data from the intersts button @Addeb you can use it later
    console.log(clickedInterests);

    return (
        <div className='mt-10 '>
            {/* <SecondaryTitle text={t("yourInterests")} /> */}
            <div className='flex flex-wrap items-center justify-center overflow-hidden lg:ml-8 lg:mr-8 lg:flex-row lg:flex-wrap lg:items-center lg:justify-center'>
                {interests.map((interest) => (
                    <SingleInterest
                        key={interest}
                        text={t(interest)}
                        onInterestClick={handleInterestClick}
                    />
                ))}
            </div>
        </div>
    );
};

export default Interests;
