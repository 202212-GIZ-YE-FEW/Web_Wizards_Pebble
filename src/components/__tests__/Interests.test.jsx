import { I18nContext } from "next-i18next";
import renderer from "react-test-renderer";

import Interests from "../Interests";

it("renders correctly", () => {
    const i18n = {
        language: "en",
        t: (str) => str,
    };
    const tree = renderer
        .create(
            <I18nContext.Provider value={i18n}>
                <Interests t={i18n.t} />
            </I18nContext.Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
