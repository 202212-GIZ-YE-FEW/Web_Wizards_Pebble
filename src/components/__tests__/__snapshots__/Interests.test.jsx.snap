// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-10 "
>
  <div>
    <label
      className="font-extrabold text-2xl py-2 ml-3 text-gray-800"
    >
      yourInterests
    </label>
  </div>
  <div
    className="flex flex-wrap justify-evenly"
  >
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            noPoverty
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            goodHealth
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            qualityEducation
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            genderEquality
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            cleanWater
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            affordableEnergy
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            decentWork
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            innovation
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            reducedInequalities
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            sustainableCities
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            responsibleConsumption
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            lifeBelowWater
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            lifeOnLand
          </p>
        </div>
      </button>
    </div>
    <div
      className="w-[33%] flex justify-center lg:py-3 lg:px-3 md:py-2 md:px-2 py-1 px-1 "
    >
      <button
        className="button__ButtonWrapper-sc-1aw98b4-0 button  w-full lg:h-18 h-16 px-1 transition bg-white "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid black",
            "borderRadius": "0.5rem",
            "boxShadow": "2px 2px #1a1a1a",
          }
        }
      >
        <div
          className="relative flex items-center justify-center "
        >
          <p
            className="block w-max font-semibold text-gray-900 text-xs py-1 "
          >
            peaceJustice
          </p>
        </div>
      </button>
    </div>
  </div>
</div>
`;
